tap_square/sync.py

```diff
@@ -33,7 +33,31 @@ def sync(config, state, catalog):
             start_time = singer.get_bookmark(state, tap_stream_id, replication_key, config['start_date'])
```

Comment:
In general this sync method needs refactoring now that we're branching 3 times. There's definitely some repeated codes in those branches.